src/validation.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const functionSchema = () => {
+  return {
+    properties: {
+      allowAccess: {
+        anyOf: [{
+          type: 'array',
+          items: {
+            type: 'string'
+          }
+        }, {
+          type: 'string'
+        }
+      ]
+      },
+    },
+  }
+}
+
+const providerSchema = () => {
+  return {
+      provider: {
+        properties: {
+          access: { type: 'object' },
+        },
+      }
+    }
+}
```

Comment:
Couldn't decide if I should include ajv to test these schemas?

Reply:
Does serverless provide AJV ?

Reply:
yeah its new as of 2 https://github.com/serverless/serverless/blob/master/package.json#L29

Reply:
But it looks like the serverless plugin repos try to avoid including serverless and you need to test in a real service to know if it works. Unsure of best practices, this is the first time I've worked on a plugin.

Reply:
I would add it as a devDep then (the same version) and add schema tests.

Reply:
> Unsure of best practices, this is the first time I've worked on a plugin.

Me either, but I don't think validating our ajv schemas is a bad idea.